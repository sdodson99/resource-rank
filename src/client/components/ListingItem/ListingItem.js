import React from 'react';
import PropTypes from 'prop-types';
import styles from './ListingItem.module.css';

const ListingItem = ({ children, hover, xPadding }) => {
  const getClassName = () => {
    let className = styles.ListingItem;

    if (hover) {
      className += ` ${styles.HoverListingItem}`;
    }

    if (xPadding) {
      className += ' px-3';
    }

    return className;
  };

  return (
    <div className={getClassName()} data-testid="ListingItem">
      {children}
    </div>
  );
};

ListingItem.propTypes = {
  children: PropTypes.node,
  hover: PropTypes.bool,
  xPadding: PropTypes.bool,
};

ListingItem.defaultProps = {
  hover: false,
  xPadding: true,
};

export default ListingItem;
