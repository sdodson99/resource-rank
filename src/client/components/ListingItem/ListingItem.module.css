.ListingItem {
  @apply py-8 px-3;
}

.HoverListingItem {
  @apply bg-transparent;
}

.HoverListingItem:hover {
  @apply bg-gray-100 transition duration-200;
}
