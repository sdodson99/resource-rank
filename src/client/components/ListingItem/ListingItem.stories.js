import React from 'react';
import ListingItem from './ListingItem';

export default {
  component: ListingItem,
  title: 'Components/ListingItem',
  decorators: [
    (Story) => {
      return <Story />;
    },
  ],
};

const Template = (args) => <ListingItem {...args} />;

export const Default = Template.bind({});
Default.args = {
  children: 'Listing item content.',
};

export const Hover = Template.bind({});
Hover.args = {
  ...Default.args,
  hover: true,
};
