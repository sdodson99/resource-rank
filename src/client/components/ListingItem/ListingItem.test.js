import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import ListingItem from './ListingItem';
import { createRenderer } from 'react-test-renderer/shallow';

describe('<ListingItem />', () => {
  it('should mount', () => {
    render(<ListingItem />);

    const listingItem = screen.getByTestId('ListingItem');

    expect(listingItem).toBeInTheDocument();
  });

  describe('with hover enabled', () => {
    it('should render correctly', () => {
      const tree = createRenderer().render(
        <ListingItem hover={true}>Hello world.</ListingItem>
      );

      expect(tree).toMatchSnapshot();
    });
  });

  describe('with hover disabled', () => {
    it('should render correctly', () => {
      const tree = createRenderer().render(
        <ListingItem hover={false}>Hello world.</ListingItem>
      );

      expect(tree).toMatchSnapshot();
    });
  });

  it('should render correctly with xPadding disabled', () => {
    const tree = createRenderer().render(
      <ListingItem xPadding={false}>Hello world.</ListingItem>
    );

    expect(tree).toMatchSnapshot();
  });
});
