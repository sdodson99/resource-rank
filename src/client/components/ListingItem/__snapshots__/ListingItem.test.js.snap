// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListingItem /> should render correctly with xPadding disabled 1`] = `
<div
  className="ListingItem"
  data-testid="ListingItem"
>
  Hello world.
</div>
`;

exports[`<ListingItem /> with hover disabled should render correctly 1`] = `
<div
  className="ListingItem px-3"
  data-testid="ListingItem"
>
  Hello world.
</div>
`;

exports[`<ListingItem /> with hover enabled should render correctly 1`] = `
<div
  className="ListingItem HoverListingItem px-3"
  data-testid="ListingItem"
>
  Hello world.
</div>
`;
